src/types/local_time.rs

```diff
@@ -0,0 +1,130 @@
+use core::fmt;
+use napi::{bindgen_prelude::BigInt, Error, Status};
+use scylla::frame::value::CqlTime;
+use std::num::ParseIntError;
+
+const NANO_SEC_IN_SEC: i64 = 1000000000;
+
+#[napi]
+pub struct LocalTimeWrapper {
+    pub value: BigInt,
+    pub hour: i64,
+    pub minute: i64,
+    pub second: i64,
+    pub nanosecond: i64,
+    ns_value: i64,
+}
+
+#[napi]
+impl LocalTimeWrapper {
+    fn convert_to_object(ns_value: i64) -> Self {
+        let mut second: i64 = ns_value / NANO_SEC_IN_SEC;
+        let nanosecond: i64 = ns_value - (second * NANO_SEC_IN_SEC);
+
+        let mut minute = second / 60;
+        let hour = minute / 60;
+
+        second %= 60;
+        minute %= 60;
+
+        LocalTimeWrapper {
+            value: BigInt::from(ns_value),
+            hour,
+            minute,
+            second,
+            nanosecond,
+            ns_value,
+        }
+    }
+
+    #[napi]
+    pub fn new(total_nanoseconds: BigInt) -> napi::Result<Self> {
+        let (ns_value, is_lossless) = total_nanoseconds.get_i64();
+
+        if !is_lossless {
+            return Err(Error::new(
+                Status::GenericFailure,
+                "Nanoseconds must not overflow i64",
+            ));
+        }
+
+        Ok(Self::convert_to_object(ns_value))
+    }
+
+    /// format: hh:MM:ss.ns
+    #[napi(js_name = "toString")]
+    pub fn to_format(&self) -> String {
+        self.to_string()
+    }
+
+    #[napi]
+    pub fn from_string(s: String) -> napi::Result<BigInt> {
+        let lambda = |s: String| -> Result<i64, ParseIntError> {
+            let parts: Vec<&str> = s.split(':').collect();
+            let mut millis: i64 = 0;
+
+            millis += parts[0].parse::<i64>()? * 3600000;
+            millis += parts[1].parse::<i64>()? * 60000;
```

Comment:
How will the code behave if parts has 1 or 0 elements?

Reply:
It appears there are no tests for this function with invalid string. Maybe add some unit tests for that.

Reply:
> How will the code behave if parts has 1 or 0 elements?

The elements 0 and 1 will always be in this string. This is checked by the regex in JS. So rust at this point will not throw an error.